providers/default.rb

```diff
@@ -201,26 +201,42 @@ def windows?
 # gem files.
 #
 def move_opt_chef(src, dest)
-  converge_by("move all files under #{src} to #{dest}") do
-    FileUtils.rm_rf dest
-    raise "rm_rf of #{dest} failed" if ::File.exist?(dest) # detect mountpoints that were not deleted
-    FileUtils.mv src, dest
-  end
-rescue => e
-  # this handles mountpoints
-  converge_by("caught #{e}, falling back to copying and removing from #{src} to #{dest}") do
+  if windows?
+    powershell_script 'move src -> using powershell' do
+      code <<-EOH
+        while ((test-path "#{src}") -And ($retryCount -lt 150)) {
+          Rename-Item "#{src}" "#{dest}"
+          $retryCount++
+          sleep 1
+        }
+        if (test-path "#{src}") { exit 1 }
+      EOH
+      action :nothing
+    end.run_action(:run)
+  else
     begin
+    converge_by("move all files under #{src} to #{dest}") do
       FileUtils.rm_rf dest
-    rescue
-      nil
-    end # mountpoints can throw EBUSY
-    begin
-      FileUtils.mkdir dest
-    rescue
-      nil
-    end # mountpoints can throw EBUSY
-    FileUtils.cp_r Dir.glob("#{src}/*"), dest
-    FileUtils.rm_rf Dir.glob("#{src}/*")
+      raise "rm_rf of #{dest} failed" if ::File.exist?(dest) # detect mountpoints that were not deleted
+      FileUtils.mv src, dest
+    end
+  rescue => e
+    # this handles mountpoints
+    converge_by("caught #{e}, falling back to copying and removing from #{src} to #{dest}") do
```

Comment:
if moving files on windows is problematic, then i think my first choice would be to avoid the use of powershell and just copy files.  correctness, simplicity and reduction of dependencies all take much higher priority over performance in this cookbook.